modules/tutorials/pages/schema-design-guide.adoc

```diff
@@ -1 +1,267 @@
+
+[source.wrap,gsql]
+----
+CREATE EDGE Works_For (FROM Person, TO Company)
+----
+
+*Reversed Edge (Parent-Child Relationship)*
+
+A *reversed edge* allows you to define both directions of a relationship.
+For example, in a *parent-child relationship*, we might define the primary edge as *"parent_of"* (Parent → Child).
+However, to easily traverse the relationship in the opposite direction (from Child to Parent), we can use a *reversed edge* with a different name, like "child_of" (Child → Parent). This makes it easy to query in either direction.
+
+image::Reversed Edge.png[]
+
+The following schema creates *reversed edges* for a *parent-child* relationship:
+
+[source.wrap,gsql]
+----
+CREATE EDGE Parent_Of (FROM Person, TO Person)
+CREATE EDGE Child_Of (FROM Person, TO Person)
+----
+
+==== Multiple Relationships Between Two Vertices
+
+For situations where different types of relationships occur between two vertices (e.g., purchases and reviews, or purchases and returns), you can represent each relationship as a discriminated edge.
+
+The figure below shows a 'Person' vertex connected to a 'Product' vertex by two different edges: 'purchased' and 'returned'. Each edge clearly indicates a different type of interaction.
+
+image::Multiple relationship between two vertices.png[]
+
+The following scheme creates discriminated edges to represent different types of *events* (e.g., *purchase and return*):
+
+[source.wrap,gsql]
+----
+CREATE EDGE Purchase_Event (FROM Person, TO Product, purchase_date DATETIME, quantity INT)
+CREATE EDGE Return_Event (FROM Person, TO Product, return_date DATETIME, reason STRING)
+----
+
+=== Indexing - Optimizing Searches
+
+Efficient querying relies on optimizing how the data is indexed. When you design a graph schema, proper indexing plays a key role in speeding up searches.
+
+==== Vertex Indexes
+
+Use vertex indexes for attributes that are queried frequently, like user ID or product ID.
+
+==== Secondary Indexes
+
+Use secondary indexes for attributes that require range queries or non-primary attributes.
+
+== Optimizing for Your Use Case
+
+=== Attributes - Property or Vertex Type
+
+When defining the properties of entities, you must decide whether to model them as attributes or vertices.
+
+=== Attributes
+Attributes are properties that are directly associated with vertices or edges (e.g., product color, transaction date).
+
+=== Vertices
+Model properties as vertices when they are frequently queried or need to be linked to multiple other vertices (e.g., color as a vertex for products).
+
+=== Granularity of Edge Types
+
+When deciding how detailed your relationships should be, you have two options: *multiple edge types* or a *single edge type with attributes*.
+
+==== Multiple Edge Types
+
+If your relationships are highly specific (e.g., "is_friend_of", "is_colleague_of"), use multiple edge types. This is memory-intensive but efficient for specific queries.
```

Comment:
```suggestion
If your relationship types are well-defined and not to numerous (e.g., "is_friend_of", "is_colleague_of"), use multiple edge types.
```
I don't think there is a memory problem. It's just hard to manage if there are too many types.